docs/llm_deployment.md

```diff
@@ -0,0 +1,73 @@
+# LLM Deployment with TorchServe
+
+This document describes how to easily serve large language models (LLM) like Meta-Llama3 with TorchServe.
+Besides a quick start guide using our VLLM integration we also provide a list of examples which describe other methods to deploy LLMs with TorchServe.
+
+## Quickstart LLM Deployment
+
+TorchServe offers easy LLM deployment through its VLLM integration.
+Through the integration of our [LLM launcher script](https://github.com/pytorch/serve/blob/7a9b145204b4d7cfbb114fe737cf980221e6181e/ts/llm_launcher.py) users are able to deploy any model supported by VLLM with a single command.
+The launcher can either be used standalone or in combination with our provided TorchServe GPU docker image.
+
+To launch the docker we first need to build it:
+```bash
+docker build . -f docker/Dockerfile.llm -t ts/llm
+```
+
+Models are usually loaded from the HuggingFace hub and are cached in a [docker volume](https://docs.docker.com/storage/volumes/) for faster reload.
+If you want to access gated models like the Meta-Llama3 model you need to provide a HuggingFace hub token:
+```bash
+export token=<HUGGINGFACE_HUB_TOKEN>
+```
+
+You can then go ahead and launch a TorchServe instance serving your selected model:
+```bash
+docker run --rm -ti --gpus all -e HUGGING_FACE_HUB_TOKEN=$token -p 8080:8080 -v data:/data ts/llm --model_id meta-llama/Meta-Llama-3-8B-Instruct --disable_token
```

Comment:
Is this needed `-e HUGGING_FACE_HUB_TOKEN=$token` Why not directly set `export HUGGING_FACE_HUB_TOKEN= <HUGGINGFACE_HUB_TOKEN>` 

Also from the security POV,  does the existing command print the token?

Reply:
AFAIK the docker will not pick up env variables from the calling environment. So you would still have 
```
export HUGGING_FACE_HUB_TOKEN= <HUGGINGFACE_HUB_TOKEN>
docker run --rm -ti --gpus all -e HUGGING_FACE_HUB_TOKEN=$HUGGING_FACE_HUB_TOKEN ...
```
which is even longer and it comes down to the same process. For a RL deployment the token variable would be set through a secret and so the token would not show up in any of the logs.